packages/@react-spectrum/button/src/Button.tsx

```diff
@@ -51,6 +51,7 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
   let {hoverProps, isHovered} = useHover({isDisabled});
   let {styleProps} = useStyleProps(otherProps);
   let hasLabel = useHasChild(`.${styles['spectrum-Button-label']}`, domRef);
+  let hasIcon = useHasChild(`.${styles['spectrum-Button-icon']}`, domRef);
```

Comment:
Add this conditional back in to make the icon-only scenario more specific because of the [Button Events example](https://react-spectrum.adobe.com/react-spectrum/Button.html#events).

In that case, React is splitting the children into a array -- [number, string], so the Button code will render that since it doesn't qualify `typeof children === 'string'`. Since it renders as is, the `.spectrum-Button-label` doesn't get applied, and with the previous code, it would apply the icon only styling, thus removing the min-width/padding.

Reply:
Seems fine. Unfortunately, you'll have to move this code change to a separate pr so we can merge it prior to release. The docs have to wait until after release.